src/Swarm/Game/Display.hs

```diff
@@ -98,14 +99,19 @@ displayPriority :: Lens' Display Priority
 invisible :: Lens' Display Bool
 
 instance FromJSON Display where
-  parseJSON = withObject "Display" $ \v ->
-    Display
-      <$> v .:? "char" .!= ' '
-      <*> v .:? "orientationMap" .!= M.empty
-      <*> v .:? "curOrientation"
-      <*> (fmap (worldPrefix <>) <$> v .:? "attr") .!= entityAttr
-      <*> v .:? "priority" .!= 1
-      <*> v .:? "invisible" .!= False
+  parseJSON v = runE (parseJSONE v) (defaultEntityDisplay ' ')
+
+instance FromJSONE Display Display where
```

Comment:
I don't understand the purpose of making this into a `FromJSONE Display Display` instance.  It seems unnecessarily complicated.  Why not just keep the simple `FromJSON Display` instance and write something like 
```
do
  let defD = defaultEntityDisplay ' '
  Display
    <$> v .:? "char" .!= (defD ^. defaultChar)
```
and so on?

Reply:
Oh, wait, never mind, I get it now, it's so that you can pass in a different default display from the `FromJSONE EntityMap TRobot` instance, depending on the value of `system`.

Reply:
The real reason is that I want something better than `FromJSONE Bool` and the only reasonable wrapper seemed to be `Display` itself. The idea is really to provide defaults and in code it seemed to work OK.